service/status-report/main.go

```diff
@@ -109,12 +109,15 @@ func main() {
 		klog.Exitf("Failed to report status of storageClient %v: %v", storageClient.Status.ConsumerID, err)
 	}
 
-	storageClientCopy := &v1alpha1.StorageClient{}
-	storageClient.DeepCopyInto(storageClientCopy)
-	if utils.AddAnnotation(storageClient, utils.DesiredSubscriptionChannelAnnotationKey, statusResponse.DesiredClientOperatorChannel) {
-		// patch is being used here as to not have any conflicts over storageclient cr changes as this annotation value doesn't depend on storageclient spec
-		if err := cl.Patch(ctx, storageClient, client.MergeFrom(storageClientCopy)); err != nil {
-			klog.Exitf("Failed to annotate storageclient %q: %v", storageClient.Name, err)
+	// statusResponse.DesiredClientOperatorChannel once ocs-o/r or merges will change this to read hash
+	if statusResponse.DesiredClientOperatorChannel != storageClient.Status.DesiredClientConfigHash {
+		storageClientCopy := &v1alpha1.StorageClient{}
+		storageClient.DeepCopyInto(storageClientCopy)
+		if utils.AddAnnotation(storageClient, utils.DesiredClientConfigAnnotationKey, statusResponse.DesiredClientOperatorChannel) {
+			// patch is being used here as to not have any conflicts over storageclient cr changes as this annotation value doesn't depend on storageclient spec
+			if err := cl.Patch(ctx, storageClient, client.MergeFrom(storageClientCopy)); err != nil {
+				klog.Exitf("Failed to annotate storageclient %q: %v", storageClient.Name, err)
+			}
```

Comment:
the idea is you only set the hash and controller takes the action if hash is changed but not split action b/n reporter & controller.